### PR TITLE
fix: update mobile web trade form for better ux experience on isolated trades

### PR DESCRIPTION
**Pull request by moo-onthelawn**
There are no designs for this yet, this is just to tide us over until ui refresh designs come in; should match what we have in web

iphone SE

| buy | sell | 
| ---- | ---- |
| Screenshot 2024-10-04 at 12 26 54 PM | Screenshot 2024-10-04 at 12 26 59 PM |

---

## Views

- New: `<TradeSideTabs>`
  - Moved to own component because it's shared by the TradeDialog + TradeBoxOrderView to avoid having to think about styling

- New: `forms/TradeForm/MarginAndLeverageButtons`
  - Moved to own component because it's shared by the TradeDialog + TradeBoxOrderView to avoid having to think about styling

- `<TradeForm>`
  - Note: `currentStep` is not null only in mobile / tablet views
  - Add fade to end of toggle group so it's more obvious it's scrollable
  - Update to render the tabs on the edit order step

- `TradeBoxOrderView`
  - No change, just update to use new `TradeSideTabs` and `MarginAndLeverageButtons` component, remove mostly unnecessary `tabsStyle`
  
## Components

- `TradeDialog`
  - Update the `EditOrder` step so that the new header is just `Trade`, and we have the `TradeSideTabs` and `MarginAndLeverageButtons` in the content of the dialog

- `Icon`
  - Update css sizing to allow it to be set from btoh css and typescript

- Delete: `TradeSideToggle`
  - No longer used

## Components

- `<Collapsible>`
  - Fix `TriggerIcon` sizing to match with `--trigger-icon-width`

- `<Icon>`
  - Clean up sizing
   
- `Tabs`
  - Use `useFadeOnHorizontalScrollContainer` to auto apply fade  

- `ToggleGroup`
  - Use `useFadeOnHorizontalScrollContainer` to auto apply fade  

## Hooks

- `useFadeOnHorizontalScrollContainer`
  - Logic for determining if we should add fade to scroll container

## Styles

- `<layoutMixins>`
  - Add `overflow: hidden` to fade mixins

### CODE REVIEW

File: src/styles/layoutMixins.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ const scrollAreaFadeStart = css`
 
   /* Rules */
   position: relative;
+  overflow: hidden;
```
Comment (tyleroooo) [inline src/styles/layoutMixins.ts:47]: gotta imagine this is going to cause pain SOMEWHERE, but I've got no clue where

File: src/views/forms/TradeForm.tsx [hunk 2]
```diff
@@ -397,6 +412,12 @@ const $TradeForm = styled.form`
   }
 `;
 
+const $Content = styled.div`
```
Comment (tyleroooo) [inline src/views/forms/TradeForm.tsx:415]: mildly prefer just inlining these short ones (except FadeContainer) with tw=

File: src/styles/layoutMixins.ts (hunk 1 above)
Comment (moo-onthelawn) [reply src/styles/layoutMixins.ts:47]: oh rip good catch - it's preventing the "More" dropdown from showing. Man lemme fix

File: src/components/ToggleGroup.tsx [hunk 3]
```diff
@@ -48,42 +48,98 @@ export const ToggleGroup = forwardRefFn(
   ) => {
     const { isTablet } = useBreakpoints();
 
+    const [showFadeStart, setShowFadeStart] = useState(false);
+    const [showFadeEnd, setShowFadeEnd] = useState(false);
+
+    const handleScroll = useCallback((e: UIEvent<HTMLElement>) => {
```
Comment (moo-onthelawn) [inline src/components/ToggleGroup.tsx:54]: potentially could move this into a hook too 👀

**Review by aforaleka — APPROVED** [review]

**Review by aforaleka — APPROVED** [review]

**Pull request merged**